### ISSUE TITLE
What is gradient accumulation?

### ISSUE THREAD

**Issue by OverLordGoldDragon** [CyberZHG/keras-gradient-accumulation#3]
What paper is it based on - do you have a link? Found nothing via search engine

**Comment by OverLordGoldDragon** [comment]
I suppose I could call an excavator to budge it, but moving it entirely would require jet fuel.

**Comment by OverLordGoldDragon** [comment]
Found it: [original paper](https://kexue.fm/archives/6794) -- [English translation](https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=https%3A%2F%2Fkexue.fm%2Farchives%2F6794)

In a nutshell: enables larger batch size training (circumventing OOM) by accumulating / buffering gradients instead of applying them directly. BERT is cited as a major application, where large batch sizes are desirable but often unachievable per memory constraints.
